content/blog/duckplyr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,209 @@
+install.packages("duckplyr")
+```
+
+In this article, we'll show how duckplyr can help you with data of different size, explain how you can help improve the package, and ... .
+
+## A drop-in replacement for dplyr
+
+The duckplyr package is a _drop-in replacement for dplyr_ that uses _DuckDB for speed_.
+You can simply _drop_ duckplyr into your pipeline by loading it, then computations will be efficiently carried out by DuckDB.
+DuckDB is a fast in-memory analytical database system[^duckdb].
+
+[^duckdb]: If you haven't heard about it, you can watch [Hannes Mühleisen's keynote at posit::conf(2024)](https://www.youtube.com/watch?v=GELhdezYmP0&feature=youtu.be).
+
+
+```{r}
+library(conflicted)
+library(duckplyr)
+conflict_prefer("filter", "dplyr", quiet = TRUE)
+library(babynames)
+
+out <- babynames |>
+  mutate(prevalence = if_else(prop >= 0.01, "frequent", "rare")) |>
+  summarize(
+    .by = c(sex, year, prevalence),
+    babies_n = sum(n)
+  ) |>
+  filter(sex == "F")
+class(out)
+out
+```
+
+Like with other dplyr backends like dtplyr and dbplyr, duckplyr allows you to get faster results.
+Unlike other dplyr backends, duckplyr does not require you to learn a different syntax.
+
+The duckplyr package is fully compatible with dplyr: if an operation cannot be carried out with DuckDB, it is automatically outsourced to dplyr.
+Over time, we expect fewer and fewer fallbacks to dplyr to be needed.
+
+## How to use duckplyr
+
+To _replace_ dplyr with duckplyr, you can:
+
+- Load duckplyr and then keep your pipeline as is. Calling `library(duckplyr)` overwrites dplyr methods, enabling duckplyr for the entire session no matter how data.frames are created.
+    This is shown in the example above.
+
+- Create individual "duck frames" using _conversion functions_ like `duckdb_tibble()` or `as_duckdb_tibble()`, or _ingestion functions_ like `read_csv_duckdb()`.
+
+
+Then, the data manipulation pipeline uses the exact same syntax as a dplyr pipeline.
+The duckplyr package performs the computation using DuckDB.
+
+```{r}
+# Undo the effect of library(duckplyr)
+methods_restore()
+
+out <- babynames |>
+  as_duckdb_tibble() |>
+  mutate(prevalence = if_else(prop >= 0.01, "frequent", "rare")) |>
+  summarize(
+    .by = c(sex, year, prevalence),
+    babies_n = sum(n)
+  ) |>
+  filter(sex == "F")
+class(out)
+```
+
+In both cases, printing the result only shows the first few rows, as with dbplyr.
+
+```{r}
+out
+```
+
+The result can finally be materialized to memory, or computed temporarily, or computed to a file.
+
+```{r}
+# to memory
+nrow(out)
+# or for instance collect(out)
+
+# to a file
+csv_file <- withr::local_tempfile()
+compute_csv(out, csv_file)
+fs::file_size(csv_file)
+```
+
+When duckplyr itself does not support specific functionality, it falls back to dplyr.
+For instance, filtering on grouped data is not supported yet, still it works thanks to the fallback mechanism.
+
+```{r}
+babynames |>
+  filter(n > 10000, .by = "name")
```

Comment:
Ideally this would use an expression where grouping matters (e.g. `filter(n == max(n), .by = c(year, sex))`.